cmd/flux/stats.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/fluxcd/flux2/internal/utils"
+	"github.com/fluxcd/flux2/pkg/printers"
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	autov1 "github.com/fluxcd/image-automation-controller/api/v1beta1"
+	imagev1 "github.com/fluxcd/image-reflector-controller/api/v1beta1"
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta2"
+	notificationv1 "github.com/fluxcd/notification-controller/api/v1beta2"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta2"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var statsCmd = &cobra.Command{
+	Use:   "stats",
+	Short: "Stats of Flux reconciles",
+	Long: `The stats command prints a report of Flux custom resources present on a cluster,
+including their reconcile status and the amount of storage used for each source type`,
+	Example: `  # Print the stats report for a namespace
+  flux stats --namespace default
+
+  #  Print the stats report for the whole cluster
+  flux stats -n A`,
+	RunE: runStatsCmd,
+}
+
+type StatsFlags struct {
+	allNamespaces bool
+}
+
+var statsArgs StatsFlags
+
+func init() {
+	statsCmd.PersistentFlags().BoolVarP(&statsArgs.allNamespaces, "all-namespaces", "A", false,
+		"list the requested object(s) across all namespaces")
```

Comment:
Maybe something like

```suggestion
		"list the statistics for objects across all namespaces")
```

which seems more accurate?